src/tbp/monty/frameworks/config_utils/config_args.py

```diff
@@ -1153,6 +1154,197 @@ class FiveLMMontySOTAConfig(FiveLMMontyConfig):
     )
 
 
+"""
+Multi-LM Config Utils.
+"""
+
+
+def make_multi_lm_flat_dense_connectivity(n_lms: int) -> Dict:
+    """Create default connectivity matrices for a multi-LM experiment.
+
+    Helper function for creating monty configs for experiments having multiple sensor
+    and learning modules, and learning modules are not hierarchically connected.
+    Also assumes each LM is connected to a single upstream sensor module in 1:1 fashion,
+    and all sensor modules are mounted on a single agent.
+
+    The results of this function are primarily intended to be included in a
+    monty config dataclass/dictionary.
+
+    Args:
+        n_lms: Number of LMs. It is assumed that the number of sensor modules (not
+            including a view finder) is equal to the number of LMs.
+
+    Returns:
+        Mapping: A dictionary with keys "sm_to_agent_dict", "sm_to_lm_matrix",
+            "lm_to_lm_matrix", and "lm_to_lm_vote_matrix".
+    """
+    # Create default sm_to_lm_matrix: all sensors are on 'agent_id_0'.
+    sm_to_agent_dict = {f"patch_{i}": f"agent_id_0" for i in range(n_lms)}
+    sm_to_agent_dict["view_finder"] = "agent_id_0"
+
+    # Create default sm_to_lm_matrix: each sensor connects to one LM.
+    sm_to_lm_matrix = [[i] for i in range(n_lms)]
+
+    # Create default lm_to_lm_matrix: no LM hierarchy.
+    lm_to_lm_matrix = None
+
+    # Create default lm_to_lm_vote_matrix: all-to-all voting.
+    lm_to_lm_vote_matrix = []
+    for i in range(n_lms):
+        lst = list(range(n_lms))
+        lst.remove(i)
+        lm_to_lm_vote_matrix.append(lst)
+
+    return {
+        "sm_to_agent_dict": sm_to_agent_dict,
+        "sm_to_lm_matrix": sm_to_lm_matrix,
+        "lm_to_lm_matrix": lm_to_lm_matrix,
+        "lm_to_lm_vote_matrix": lm_to_lm_vote_matrix,
+    }
+
+
+def make_multi_lm_monty_config(
+    n_lms: int,
+    *,
+    monty_class: type,
+    learning_module_class: type,
+    learning_module_args: Optional[Mapping],
+    sensor_module_class: type,
+    sensor_module_args: Optional[Mapping],
+    motor_system_class: type,
+    motor_system_args: Optional[Mapping],
+    monty_args: Optional[Union[Mapping, MontyArgs]],
+    connectivity_func: Callable[[int], Mapping] = make_multi_lm_flat_dense_connectivity,
+) -> MontyConfig:
+    """Create a monty config for multi-LM experiments.
+
+    Creates a complete monty config for a multi-LM experiment. It is assumed that
```

Comment:
Is this still true now that you pass the `connectivity_func` as an argument?

Reply:
Ah good point. I'm deleting this bit.